src/robot/running/bodyrunner.py

```diff
@@ -596,9 +602,14 @@ def _run_finally(self, data, run):
 class WhileLimit:
 
     @classmethod
-    def create(cls, limit, variables):
+    def create(cls, limit, on_limit_message, variables):
         if not limit:
-            return IterationCountLimit(DEFAULT_WHILE_LIMIT)
+            return IterationCountLimit(DEFAULT_WHILE_LIMIT,
+                                       on_limit_message
+                                       )
+        if on_limit_message:
+            on_limit_message = variables.replace_string(
+                on_limit_message)
```

Comment:
Variables should be replaced in the beginning of `create`. Now the `IterationCountLimit` returned above gets wrong message.